### PR TITLE
[mvn] Handle recursive modules

### PR DESCRIPTION
**Pull request by juergenhoetzel**
Don't assume pom.xml is always stored in init_work_dir/MODULE.
Use directory of the parent pom.xml instead.

Fixes  reading of invalid pom.xml filenames when using > 1 levels of modules. [Apache Beam](https://github.com/apache/beam):

> infer  -g --print-logs --  mvn -pl 'org.apache.beam:beam-sdks-java-io-jdbc' compile
> Infer version v0.10.0-1580e5b3
> Copyright 2009 - present Facebook. All Rights Reserved.
> INFER_ARGS = -g^--print-logs
> rest arg: compile
> rest arg: org.apache.beam:beam-sdks-java-io-jdbc
> rest arg: -pl
> rest arg: mvn
> Project root = /home/juergen/java/beam
> CWD = /home/juergen/java/beam
> Driver mode:
> Maven driver mode:
> prog = mvn
> Arg: -pl
> Arg: org.apache.beam:beam-sdks-java-io-jdbc
> Arg: compile
> 
> Capturing in maven mode...
> Adding maven module /home/juergen/java/beam/sdks/java/build-tools
> Adding maven module /home/juergen/java/beam/sdks
> Adding maven module /home/juergen/java/beam/runners
> Adding maven module /home/juergen/java/beam/examples
> Adding maven module /home/juergen/java/beam/sdks/java/javadoc
> Adding maven module /home/juergen/java/beam/java
> Adding maven module /home/juergen/java/beam/java8
> Uncaught exception:
>   
>   (Sys_error
>    "/home/juergen/java/beam/java8/pom.xml: No such file or directory")
> 
> Raised by primitive operation at file "pervasives.ml", line 335, characters 20-46
> Called from file "src/in_channel.ml", line 14, characters 2-30
> Called from file "integration/Maven.ml", line 133, characters 2-49
> Called from file "integration/Maven.ml", line 148, characters 4-39
> Called from file "backend/infer.ml", line 455, characters 2-21
>  

Using this pull request:

> Infer version v0.10.0-1580e5b3
> Copyright 2009 - present Facebook. All Rights Reserved.
> INFER_ARGS = -g^--print-logs
> rest arg: compile
> rest arg: org.apache.beam:beam-sdks-java-io-jdbc
> rest arg: -pl
> rest arg: mvn
> Project root = /home/juergen/java/beam
> CWD = /home/juergen/java/beam
> Driver mode:
> Maven driver mode:
> prog = mvn
> Arg: -pl
> Arg: org.apache.beam:beam-sdks-java-io-jdbc
> Arg: compile
> 
> Capturing in maven mode...
> Adding maven module /home/juergen/java/beam/sdks/java/build-tools
> Adding maven module /home/juergen/java/beam/sdks
> Adding maven module /home/juergen/java/beam/runners
> Adding maven module /home/juergen/java/beam/examples
> Adding maven module /home/juergen/java/beam/sdks/java/javadoc
> Adding maven module /home/juergen/java/beam/examples/java
> Adding maven module /home/juergen/java/beam/examples/java8
> Adding maven module /home/juergen/java/beam/runners/core-construction-java
> Adding maven module /home/juergen/java/beam/runners/core-java
> Adding maven module /home/juergen/java/beam/runners/direct-java
> Adding maven module /home/juergen/java/beam/runners/flink
> Adding maven module /home/juergen/java/beam/runners/google-cloud-dataflow-java
> Adding maven module /home/juergen/java/beam/runners/spark
> Adding maven module /home/juergen/java/beam/runners/apex
> Adding maven module /home/juergen/java/beam/runners/flink/runner
> Adding maven module /home/juergen/java/beam/runners/flink/examples
> Adding maven module /home/juergen/java/beam/sdks/common
> Adding maven module /home/juergen/java/beam/sdks/java
> Adding maven module /home/juergen/java/beam/sdks/python
> Adding maven module /home/juergen/java/beam/sdks/java/core
> Adding maven module /home/juergen/java/beam/sdks/java/io
> Adding maven module /home/juergen/java/beam/sdks/java/maven-archetypes
> Adding maven module /home/juergen/java/beam/sdks/java/extensions
> Adding maven module /home/juergen/java/beam/sdks/java/harness
> Adding maven module /home/juergen/java/beam/sdks/java/java8tests
> Adding maven module /home/juergen/java/beam/sdks/java/extensions/join-library
> Adding maven module /home/juergen/java/beam/sdks/java/extensions/sorter
> Adding maven module /home/juergen/java/beam/sdks/java/maven-archetypes/starter
> Adding maven module /home/juergen/java/beam/sdks/java/maven-archetypes/examples
> Adding maven module /home/juergen/java/beam/sdks/java/maven-archetypes/examples-java8
> Adding maven module /home/juergen/java/beam/sdks/java/io/elasticsearch
> Adding maven module /home/juergen/java/beam/sdks/java/io/google-cloud-platform
> Adding maven module /home/juergen/java/beam/sdks/java/io/hadoop-common
> Adding maven module /home/juergen/java/beam/sdks/java/io/hbase
> Adding maven module /home/juergen/java/beam/sdks/java/io/hdfs
> Adding maven module /home/juergen/java/beam/sdks/java/io/jdbc
> Adding maven module /home/juergen/java/beam/sdks/java/io/jms
> Adding maven module /home/juergen/java/beam/sdks/java/io/kafka
> Adding maven module /home/juergen/java/beam/sdks/java/io/kinesis
> Adding maven module /home/juergen/java/beam/sdks/java/io/mongodb
> Adding maven module /home/juergen/java/beam/sdks/java/io/mqtt
> Adding maven module /home/juergen/java/beam/sdks/common/fn-api
> Adding maven module /home/juergen/java/beam/sdks/common/runner-api
> Running maven capture:
>

### CODE REVIEW

File: infer/src/integration/Maven.ml [hunk 1]
```diff
@@ -10,6 +10,7 @@ open! IStd
 
 module CLOpt = CommandLineOption
 module L = Logging
+module F = Filename
```
Comment (jvillard) [inline infer/src/integration/Maven.ml]: The (completely undocumented) style in infer is to abbreviate some modules we commonly use, but `Filename` is not one of them, and `F` stands for `Format` elsewhere in the code. I'll try and document this.

tl; dr: please remove this line and use the full module name below instead of `F`.

**Comment by juergenhoetzel** [comment]
> So that this functionality doesn't regress in the future, could you change this test so that it has recursive submodules?

@jvillard Thanks for your feedback. I added an intermediate `parent pom` for `module2`. 

I also reordered my commits (failing testcase before fix).
